Make tplink SmartStrip communication more robust

gjbadros opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->


## Proposed change
<!-- 
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
In get_static_devices, protect the execution from a failed/inaccessible SmartStrip. Before this change, one smartstrip throwing an exception would break all static devices from the integration.

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box! 
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example entry for `configuration.yaml`:
<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR. Furthermore, for new integrations, it gives an impression of how
  the configuration would look like.
  Note: Remove this section if this PR does not have an example entry.
-->

```yaml
# Example configuration.yaml

```

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [X] The code change is tested and works locally.
- [X] Local tests pass. **Your PR cannot be merged unless tests pass**
- [X] There is no commented out code in this PR.
- [X] I have followed the [development checklist][dev-checklist]
- [X] The code has been formatted using Black (`black --fast homeassistant tests`)
- [X] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

gjbadros committed:
In get_static_devices, protect other devices if a SmartStrip constructor throws an exception -- this happens to me lots and breaks all my devices in this integration unnecessarily.

gjbadros committed:
In get_static_devices, protect other devices if a SmartStrip constructor throws an exception -- this happens to me lots and breaks all my devices in this integration unnecessarily.

rytilahti commented:
#30719 will revamp the whole integration and should fix this issue, too.

gjbadros commented:
Ok, cool -- I trust that's coming soon, if not, it might not hurt to get this tiny fix in while we wait. No big deal either way, though... I have the patch to reapply to my own as I upgrade.

rytilahti commented:
If you are feeling adventurous, you could go and check out https://github.com/python-kasa/python-kasa/pull/59 in combination with https://github.com/home-assistant/core/pull/30719 . It should work much better due to fewer I/O reqs (e.g., on HS300 the number of reqs should go down from 19 to 1 per update cycle).

gjbadros commented:
Will do, if time permits -- sounds like a great improvement and I'm looking
forward to using the new implementation... thanks!

On Sat, May 16, 2020 at 2:58 PM Teemu R. <notifications@github.com> wrote:

> If you are feeling adventurous, you could go and check out
> python-kasa/python-kasa#59
> <https://github.com/python-kasa/python-kasa/pull/59> in combination with
> #30719 <https://github.com/home-assistant/core/pull/30719> . It should
> work much better due to fewer I/O reqs (e.g., on HS300 the number of reqs
> should go down from 19 to 1 per update cycle).
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/home-assistant/core/pull/35628#issuecomment-629711285>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AALOHTOR7IYGZ7XG5Q25RZDRR4EA5ANCNFSM4NA4DRXA>
> .
>

Commit:
Fix devolo sensor subscriber (#37337)

bsmappee committed:
Smappee dependency update (#37331)

jhollowe committed:
Update proxmoxve integration to correctly renew authentication (#37016)

hdsheena committed:
Change log url in config check error notification (#37311)

vangorra committed:
Fix gogogate2 issue where non-admin users could not login (#37353)

uvjustin committed:
Use entry.data.get() in forked_daapd config_flow as some entries miss… (#37359)

bramkragten committed:
Update frontend to 20200702.0 (#37369)

vangorra committed:
Fix withings bug that grabbed oldest value instead of the newest (#37362)

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

starkillerOG committed:
Add DenonAvr missing error message (#37370)

starkillerOG committed:
Add DenonAvr missing error message (#37370)

balloob committed:
Bumped version to 0.112.1

rytilahti commented:
We could get this in, if it helps for the blocking issues described in #34465; Failing to show some entities is a smaller nuisance than blocking the whole startup.

balloob committed:
Merge pull request #37377 from home-assistant/rc

bdraco committed:
Improve unifi device tracker performance (#37308)

* Improve unifi device tracker performance

The unifi websocket sends an update every second
which generates a significant amount of state
changed updates.

Avoid creating callback functions when they
are not going to be used.

* make _no_heartbeat/_make_disconnected instance methods

* remove extra empty line

* revert is_wired change

* remove extra line

homeassistant committed:
[ci skip] Translation update

MartinHjelmare committed:
Fix ozw garage door methods (#37374)

mdegat01 committed:
Modified Influx tests to mock test queries with accurate output (#37315)

bdraco committed:
Ensure logbook entries appear when the logbook.log (#37388)

service without a domain or entity_id

bachya committed:
Bump pytile to 4.0.0 (#37398)

robbiet480 committed:
Remove my codeownership over things I dont use anymore (#37401)

elupus committed:
Convert rfxtrx tests to pytest async tests and re-enable (#37206)

* Rework and re-enable rfxtrx tests

* Add missed change to _signal_event

* Fixup the dummy serial that causes max cpu

* Make sure we cleanup thread here too

* Make sure we always wait for tasks before we check state

* Some more places we need to wait before checking

therealryanbonham committed:
Upgrade sonar to 0.2.3 (#37392)

Co-authored-by: Ryan Bonham <ryan@transparent-tech.com>

frenck committed:
Add GitHub Actions for CI (#37419)

* Add GitHub Actions for CI

* Add base/full to step descriptions

bdraco committed:
Reduce time to run zha discover tests (#37424)

Most of the time was registering services and patching
the clusters which are never calls in these tests

emontnemery committed:
Don't print MQTT credentials to log (#37364)

bdraco committed:
Ensure async_setup is mocked in geonetnz intergration tests (#37426)

* Ensure async_setup is mocked in geonetnz intergration tests

* s/asynctest/tests.async_mock/g

eifinger committed:
Add helpers.location.coordinates (#37234)

balloob committed:
Replace asynctest with tests.async_mock (#37428)

bdraco committed:
Prevent verisure lock from looping forever and sleeping in test (#37425)

ktnrg45 committed:
Fix undesired power toggling (#37427)

balloob committed:
Merge remote-tracking branch 'origin/master' into dev

bdraco committed:
Fix unmocked setup in garmin_connect test (#37429)

bdraco committed:
Fix unmocked setup in ipp tests (#37430)

balloob committed:
Stub out ecobee aux heat services (#37423)

frenck committed:
Fix building of Python Wheels (#37433)

bdraco committed:
Mock setup in sonarr config flow tests (#37432)

prystupa committed:
Add more unit tests for plum_lightpad (#37275)

* add more unit tests for plum_lightpad

* add more unit tests for plum_lightpad

* add more unit tests for plum_lightpad

* add more unit tests for plum_lightpad

* Update tests/components/plum_lightpad/test_init.py

apply suggested way to invoke async_setup_entry

Co-authored-by: Chris Talkington <chris@talkingtontech.com>

* Update tests/components/plum_lightpad/test_init.py

apply suggested way to invoke async_setup_entry

Co-authored-by: Chris Talkington <chris@talkingtontech.com>

* Update tests/components/plum_lightpad/test_init.py

apply suggested way to invoke async_setup_entry

Co-authored-by: Chris Talkington <chris@talkingtontech.com>

* Update tests/components/plum_lightpad/test_init.py

remove now unused import

Co-authored-by: Chris Talkington <chris@talkingtontech.com>

* Update tests/components/plum_lightpad/test_init.py

import Mock from tests.async_mock as suggested

Co-authored-by: Chris Talkington <chris@talkingtontech.com>

* Update tests/components/plum_lightpad/test_init.py

remove now unused import

Co-authored-by: Chris Talkington <chris@talkingtontech.com>

* Update tests/components/plum_lightpad/test_init.py

Co-authored-by: Chris Talkington <chris@talkingtontech.com>

* fix unit tests that were failing after suggested changes

Co-authored-by: Chris Talkington <chris@talkingtontech.com>

bouwew committed:
Add Plugwise zeroconf discovery (#37289)

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>
Co-authored-by: Tom Scholten <tom@sue.nl>
Co-authored-by: Tom <CoMPaTech@users.noreply.github.com>

alandtse committed:
Bump teslajsonpy to 0.9.2 (#37434)

* Bump teslajsonpy to 0.9.1
closes #37340

* Bump teslajsonpy to 0.9.2

ctalkington committed:
Mock setup in directv config flow tests (#37439)

* mock setup in directv config flow tests

* Update test_config_flow.py

rytilahti committed:
Bump python-miio to 0.5.2.1 (#37422)

bdraco committed:
Handle index already existing on db migration with MySQLdb backend (#37384)

_create_index needed the same check as _add_columns since
the MySQLdb backend throws OperationalError instead
of InternalError in this case

ctalkington committed:
Apply some suggestions from poolsense code review (#37440)

smugleafdev committed:
Fix extremely minor typo: Cosumption -> Consumption (#37322)

RogerSelwyn committed:
Fix DarkSky spamming the log (#37421)

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

nkgilley committed:
Upgrade python-join-api to allow user to specify actions (#37394)

Commit:
Use a more detailed battery icon for Tesla cars (#37154)

Use the icon_for_battery_level helper to select a battery icon that indicates charge level and whether or not the battery is actively charging.

homeassistant committed:
[ci skip] Translation update

bdraco committed:
Avoid selecting the states created column for history (#37450)

We never use the data for history.  We should
not select it as it created unneeded overhead.

bdraco committed:
Ensure removed entities are not displayed in logbook (#37395)

ctalkington committed:
Use device class to isolate tesla battery icon (#37446)

scop committed:
Remove pytest-xdist from tox now that it's in requirements_test.txt (#37455)

CoMPaTech committed:
Fix Plugwise zeroconf discovery formatting (#37457)

starkillerOG committed:
Xiaomi Gateway subdevice support & AqaraHT + SensorHT devices (#36539)

* Gateway subdevice support & AqaraHT + SensorHT devices

* Gateway subdevice support & AqaraHT + SensorHT devices

* Add starkillerOG to codeowners

as proposed by @rytilahti in this issue: https://github.com/home-assistant/core/issues/36516

* add starkillerOG to xiaomi_miio

* fix config flow tests

* Update CODEOWNERS

* Update manifest.json

* prosess revieuw comments

* fix missing import

* use proper pressure unit hPa

* subdevice --> sub_device

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* subdevice --> sub_device

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* use key acces instead of get

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* subdevice --> sub_device

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* subdevice --> sub_device

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* subdevice --> sub_device

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* subdevice --> sub_device

* use dataclass instead of namedtuple

* update to newest python-miio functions (not yet released)

* Move device info to entitie

* remove unused variable

* improve default names

* SensorHT does not support pressure

* bump python-miio to 0.5.2

* bump python-miio to 0.5.2

* bump python-miio to 0.5.2

* Fix missing brackets

Co-authored-by: Teemu R. <tpr@iki.fi>

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>
Co-authored-by: Teemu R. <tpr@iki.fi>

bdraco committed:
Fix entity_component test flapping (#37445)

frenck committed:
Fix flapping geo_json_events tests (#37471)

emontnemery committed:
Fix geonetnz_quakes test flapping (#37473)

timvancann committed:
Call sync function from async context (#37324)

emontnemery committed:
Support empty output of MQTT binary_sensor value_template (#37420)

* Support empty output of MQTT binary_sensor value_template

* Strip white space

* Add test

* Add test

emontnemery committed:
Support multiple MQTT availability topics (#37418)

* Support multiple MQTT availability topics

* Make availability list and availability_topic exclusive

* Make availability list and availability_topic exclusive

* Add missing abbreviation

emontnemery committed:
Publish birth and will messages by default (#37371)

* Publish birth and will messages by default

* Remove useless copy

frenck committed:
Fix flapping google_assistant tests (#37480)

crazyfx1 committed:
Fix base class for ViCare binary sensor to remove warning (#37478)

Shulyaka committed:
Add humidifier support to emulated_hue (#37110)

* Add humidifier support to emulated_hue

* add humidifier dependency

* move dependency to after_dependency

* move dependency to hassfest

frenck committed:
GitHub Actions: Add hadolint problem matcher (#37494)

frenck committed:
GitHub Actions: Add codespell problem matcher (#37487)

* GitHub Actions: Add codespell problem matcher

* Add some spelling issues for test

* Disable color, might throw off matcher

* Revert "Add some spelling issues for test"

This reverts commit 3afb59c8d9c4080621fc90d25d228fa308b80a4b.

frenck committed:
GitHub Actions: Add json problem matcher (#37490)

* GitHub Actions: Add json problem matcher

* Create a problem for testing

* Revert "Create a problem for testing"

This reverts commit d7735e4af39a18cb8d080bfc01e483065bbd06b6.

frenck committed:
GitHub Actions: Add pylint problem matcher (#37463)

* GitHub Actions: Add pylint problem matcher

* Create a pylint issue to test

* Create another pylint issue to test

* Register problem matcher in same step

* Apply possible workaround

* Disable problem matcher to catch raw output

* Trying again with new CI containers

* Extend problem matcher with errors and warnings

* Improve matching, keep error code in message

* Revert "Create another pylint issue to test"

This reverts commit a90e23656eaddd4f1c2995572cc39a0d41022a5f.

* Revert "Create a pylint issue to test"

This reverts commit 9dd5148eb4ef74c3261144426d1b0b0f337cb3e5.

* Fix stable name in job description

homeassistant committed:
[ci skip] Translation update

frenck committed:
GitHub Actions: Add check executables problem matcher (#37488)

* GitHub Actions: Add check executables problem matcher

* Create a problem for testing

* Revert "Create a problem for testing"

This reverts commit 3532b3777f0100316b154d334b513fe4cd62ce25.

frenck committed:
GitHub Actions: Add mypy problem matcher (#37485)

frenck committed:
GitHub Actions: Add yamllint problem matcher (#37468)

* GitHub Actions: Add yamllint problem matcher

* Introduce YAML issue to test problem matcher

* Revert "Introduce YAML issue to test problem matcher"

This reverts commit fa88c9484efc8f979fe2e3aed2f964381e319a56.

frenck committed:
GitHub Actions: Add flake8 problem matcher (#37465)

frenck committed:
GitHub Actions: Show diff on failure (#37461)

bdraco committed:
Fix flapping geonetnz_volcano test (#37497)

scop committed:
Upgrade flake8 to 3.8.3 (#37501)

https://flake8.pycqa.org/en/latest/release-notes/3.8.2.html
https://flake8.pycqa.org/en/latest/release-notes/3.8.3.html

scop committed:
Use package constraints in tox lint (#37500)

Otherwise fails with
pip._vendor.pkg_resources.ContextualVersionConflict: (importlib-metadata
1.7.0 (.../.tox/lint/lib/python3.7/site-packages),
Requirement.parse('importlib-metadata==1.6.0'), {'homeassistant'})

SeanPM5 committed:
Fix default icon for Withings sleep sensor (#37502)

The icon for Withings sleep sensor was using `mdi:bed` (which is correct for MDI v5.0.45 and later). However Home Assistant still uses an older version of Material Design Icons (4.9.95), so this `mdi:bed` icon was not displaying at all.  It should be`mdi:hotel` instead, which you can see here: https://cdn.materialdesignicons.com/4.9.95/

dcnielsen90 committed:
Fix braviatv authentication refresh (#37482)

- Bumps bravia-tv lib to 1.0.6 which fixes is_connected() to actually
      return True only when API is connected, instead of just returning whether
      or not cookies are cached (regardless if they actually worked).
    - Wrap is_connected() because it now performs io.
    - Remove unnecessary logic to refresh cookies. Now that
      is_connected() works, the bravia instance only needs to be
      reconnected when is_connected is False and TV is not off.

frenck committed:
GitHub Actions: Add pytest problem matcher (#37508)

pnbruckner committed:
Enhance automation integration to use new features in script helper (#37479)

jurgenhaas committed:
Do not count netdata cleared and undefined alarms as warnings (#37505)

* Add NetdataAlarms alarm sensor to additionally collect alarm information from the netdata host

* Incorporate suggested changes by @MartinHjelmare

* Change from pictures to icons

* Simplify API callbacks following home-assistant-ecosystem/python-netdata/pull/5 and home-assistant-ecosystem/python-netdata/pull/6

* Bring back lost empty line

* Update the library version in manifest.json

* Update the library version in requirements_all.txt

* Linting

* Linting

* Fix typo

* Do not count cleared and undefined alarms as warnings

* Do not count cleared, undefined and uninitialized alarms as warnings

michaeldavie committed:
Bump env_canada to 0.1.0 (#37483)

bdraco committed:
Prebake common history queries (#37496)

* Prebake common history queries

The python overhead of to construct
the queries exceeded the database overhead.  We now
prebake the queries that get frequently polled.

This reduces the time it takes to update history_stats
sensors and can make quite a difference if there
are a lot of them.

When using the mini-graph-card card, all the entities
on the card being graphed are queried every few seconds
for new states. Previously this would tie up the database if there
are lot of these graphs in the UI.

* Update homeassistant/components/history/__init__.py

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

* Update homeassistant/components/history/__init__.py

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

* cache entity_filter in the lambda

* switch to yield

* Revert "switch to yield"

This reverts commit f8386f494002178729b67b54dd299affd406f2f2.

* get_states always returns a list

* query wasnt actually reusable so revert part of the breakout

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

knyar committed:
Add prometheus metric naming guidelines (#37149)

* Add prometheus metric naming guidelines.

* Update homeassistant/components/prometheus/README.md

Co-authored-by: Martin Weinelt <mweinelt@users.noreply.github.com>

Co-authored-by: Martin Weinelt <mweinelt@users.noreply.github.com>

bdraco committed:
Use the main event loop for homekit (#37441)

* use hass.loop for homekit

* update calls for upstream pr

* Bump HAP-python to 2.9.2

starkillerOG committed:
Add denonavr solution tip for connection_error (#37405)

* denonavr: add solution tip for connection_error

see https://github.com/home-assistant/core/issues/37351#issuecomment-653356452

* Update en.json

* Update strings.json

shermdog committed:
Fix Datadog boolean metrics (#37273)

Explicitly cast boolean metrics to integers

bdraco committed:
Fix flapping demo geo_location test (#37516)

RogerSelwyn committed:
Standardise geniusheub error levels (#37512)

RogerSelwyn committed:
Standardis asuswrt error message level (#37515)

JeffLIrion committed:
Convert Android TV integration to async (#37510)

* Convert Android TV integration to async

* pylint

* Remove unused test code

* Require async versions of androidtv and adb-shell

* Cleanup

* Remove commented out code

* Use constants SHELL_RESPONSE_OFF and SHELL_RESPONSE_STANDBY

outadoc committed:
Add config flow + async support for SmartHab integration (#34387)

* Setup barebones SmartHab config flow

* Setup authentication flow

* Make setup async, add config flow receivers

* Add French translation

* Fix async issues

* Address review comments (thanks bdraco!)

* Fix unloading entries

* Migrate translations dir according to warning

* Create list of components

* Fix pylint false positive

* Fix bad copy-pastes 🤭

* Add async support to SmartHab component

* Address review comments (bdraco)

* Fix pylint

* Improve exception handling (bdraco)

* Apply suggestions from code review (bdraco)

Co-authored-by: J. Nick Koston <nick@koston.org>

* Don't log exceptions manually, fix error

* Reduce repeated lines in async_step_user (bdraco)

* Remove useless else (pylint)

* Remove broad exception handler

* Create strings.json + remove fr i18n

* Write tests for smarthab config flow

* Test import flow

* Fix import test

* Update homeassistant/components/smarthab/config_flow.py

Co-authored-by: J. Nick Koston <nick@koston.org>

Co-authored-by: J. Nick Koston <nick@koston.org>

jfearon committed:
Tado climate entity timer service (#37472)

* New entity service to allow timer to be set on tado climate entities

* Add new line to end of yaml

bdraco committed:
Use async_track_state_change_event for automation numeric_state (#37255)

Calling async_track_state_change_event directly
is faster than async_track_state_change (see #37251) and has
slightly lower latency triggering state updates

starkillerOG committed:
Fix xiaomi_miio error when no sensors present (#37531)

elupus committed:
Switch tests to use hass objects instead of direct (#37530)

* Switch tests to use hass objects instead of direct

* Make sure sensor update state

* Add some initial binary sensor tests

* Add initial binary sensor tests

* Add tests for pt2262

* Add test for off delay

frenck committed:
Log lines do not end with a full stop (#37527)

frenck committed:
Fix flake8 problem matcher to handle fatals as errors (#37536)

bachya committed:
Transition Guardian to use a DataUpdateCoordinator (#37380)

* Migrate Guardian to use the DataUpdateCoordinator

* Finish work

* Cleanup

* Don't use UpdateFailed error

* Code cleanup

* Code cleanup

* Remove unnecessary change

* Code review

* Code review

* Use a subclass of DataUpdateCoordinator

* Make sure to pop client upon unload

* Adjust coverage

elupus committed:
Switch rfxtrx to dispatcher (#37271)

* Switch to dispatcher

* Adjust tests for dispatcher

* Store device in sensor

* Move state application into binary sensor entity class

* Move more specifics into per platform classes

* Should not apply event in init of sensor

* Switch to call_later

* Make apply_event public

* No point in slugifying debug logs

* Adjust error in off delay and event

* Make sure we match with masked id

bdraco committed:
Cleanup async_track_state_change and augment docstring (#37251)

* Cleanup async_track_state_change and augment docstrings.

Skip from_state and to_state matching in
async_track_state_change when they are None

Optimize the state change listener for the most
common use case: no to_state and from_state
matching.

* Update benchmark to be more realistic (previously we assumed only one entity was present in the whole instance)

* Add more tests to ensure behavior is preserved

* Ensure new behavior matches test

* remove MATCH_ALL from zone automation since its the default anyways

* Might as well use async_track_state_change_event instead since MATCH_ALL is removed

pnbruckner committed:
Add Amcrest audio_detected binary sensor (#37486)

homeassistant committed:
[ci skip] Translation update

prystupa committed:
Add new integration for Bond hub (#37477)

* create foundation for Bond integration

* add Bond hub integration (fix lint)

* Update homeassistant/components/bond/__init__.py

adding async_unload_entry per PR review suggestion

Co-authored-by: Chris Talkington <chris@talkingtontech.com>

* add Bond hub integration (fix missing import after applying PR suggestion)

* Update tests/components/bond/test_init.py

add a unit for unloading per PR review suggestion

Co-authored-by: Chris Talkington <chris@talkingtontech.com>

* Update tests/components/bond/test_init.py

add unit test for unload per PR review suggestion

Co-authored-by: Chris Talkington <chris@talkingtontech.com>

* Update tests/components/bond/test_init.py

PR review suggestion

Co-authored-by: Chris Talkington <chris@talkingtontech.com>

* Update tests/components/bond/test_init.py

PR review suggestion

Co-authored-by: Chris Talkington <chris@talkingtontech.com>

* Update tests/components/bond/test_init.py

PR review suggestion

Co-authored-by: Chris Talkington <chris@talkingtontech.com>

* add Bond hub integration (fix formatting)

* Update homeassistant/components/bond/manifest.json

PR suggestion

Co-authored-by: Chris Talkington <chris@talkingtontech.com>

* Update homeassistant/components/bond/manifest.json

PR suggestion

Co-authored-by: Chris Talkington <chris@talkingtontech.com>

* Update homeassistant/components/bond/manifest.json

PR suggestion

Co-authored-by: Chris Talkington <chris@talkingtontech.com>

* Update homeassistant/components/bond/strings.json

PR suggestion

Co-authored-by: Chris Talkington <chris@talkingtontech.com>

* Update homeassistant/components/bond/manifest.json

PR suggestion

Co-authored-by: Chris Talkington <chris@talkingtontech.com>

* Update requirements_all.txt

PR suggestion

Co-authored-by: Chris Talkington <chris@talkingtontech.com>

* Update homeassistant/components/bond/manifest.json

PR suggestion

Co-authored-by: Chris Talkington <chris@talkingtontech.com>

* Update requirements_test_all.txt

PR suggestion

Co-authored-by: Chris Talkington <chris@talkingtontech.com>

* add Bond hub integration (remove friendly name from config per PR suggestion)

* Update homeassistant/components/bond/__init__.py

add per PR review feedback

Co-authored-by: Chris Talkington <chris@talkingtontech.com>

* Update homeassistant/components/bond/__init__.py

remove per PR review feedback

Co-authored-by: Chris Talkington <chris@talkingtontech.com>

* Update tests/components/bond/test_init.py

fix unit test

Co-authored-by: Chris Talkington <chris@talkingtontech.com>

Co-authored-by: Chris Talkington <chris@talkingtontech.com>

bdraco committed:
Switch homekit to use async_track_state_change_event (#37253)

* Switch homekit to use async_track_state_change_event

Calling async_track_state_change_event directly
is faster than async_track_state_change and has
slightly lower latency triggering state updates
in homekit

* check for deleted entities

* Update additional tests for linked sensor removals

* Ensure removing entities does not result in an exception

ludeeus committed:
Add missing manifest object to the check (#37535)

frenck committed:
Add dependabot for automatic updates to GitHub Actions (#37550)

dependabot[bot] committed:
Bump actions/upload-artifact from v1 to v2.0.1 (#37555)

Bumps [actions/upload-artifact](https://github.com/actions/upload-artifact) from v1 to v2.0.1.
- [Release notes](https://github.com/actions/upload-artifact/releases)
- [Commits](https://github.com/actions/upload-artifact/compare/v1...97b7dace6c8d860ce9708aba808be6a2ee4cbc3a)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump codecov/codecov-action from v1 to v1.0.10 (#37556)

Bumps [codecov/codecov-action](https://github.com/codecov/codecov-action) from v1 to v1.0.10.
- [Release notes](https://github.com/codecov/codecov-action/releases)
- [Commits](https://github.com/codecov/codecov-action/compare/v1...f3570723ef743f6942b6a480461ed0cd6c0f9baa)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

pvizeli committed:
Wheels for Python 3.8 / Alpine 3.12

scop committed:
Strings capitalization consistency fixes (#37454)

bramkragten committed:
Update frontend to 20200702.1 (#37566)

bdraco committed:
Suppress spurious homekit warning about media player sources when the device is off (#37567)

bdraco committed:
Ensure homekit tv names can be saved (#37571)

starkillerOG committed:
Ignore HEOS 1, 3, 5 and 7 for DenonAvr ssdp discovery (#37579)

jjlawren committed:
Fix Plex client controls when connected via plex.tv resource (#37572)

emontnemery committed:
Fix base topic for 'topic' (#37475)

engrbm87 committed:
Stop Speedtest sensors update on startup if manual option is enabled (#37403)

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

balloob committed:
Protect loop set default executor (#37438)

Co-authored-by: J. Nick Koston <nick@koston.org>

akloeckner committed:
Add optimistic mode to template switch (#31637)

balloob committed:
Fix loop exception handler

homeassistant committed:
[ci skip] Translation update

bdraco committed:
Mock setup in plex test to prevent CI failure (#37590)

bdraco committed:
Ensure homekit accessory reset only affect the bridges with the accessory (#37588)

prystupa committed:
apply small feedback suggestions from a previous PR that is already merged (#37551)

balloob committed:
Bump voluptuous-serialize 2.4.0 (#37241)

* Bump voluptuous-serialize 2.4.0

* Add constant to demo

* gen reqs

jjlawren committed:
Fix missing Plex account mocks in tests (#37591)

bieniu committed:
Bump backend library for Dune HD integration (#37594)

DarkFox committed:
Update Rejseplanen rjpl to 0.3.6 (#37215)

pvizeli committed:
Update azure-pipelines-wheels.yml for Azure Pipelines

bdraco committed:
Make devices and activities visible as harmony attributes (#37559)

* Make devices and activities visibile as harmony attributes

* Allow restoring previous activity, add tests

* fix test

* Kill activity_notify with fire

* remove trailing ,

bsmappee committed:
Place smappee supported configurations after device class (#37595)

starkillerOG committed:
Fix DenonAvr discovery of Denon DN-500AV (#37529)

* fix discovery of Denon DN-500AV

* fix discovery of Denon DN-500AV

* fix discovery of Denon DN-500AV

* fix typo

* bump denonavr to 0.9.4

* add ignored model test

* fix mistake in branch

basnijholt committed:
Bump aiokef to v0.2.13 which fixes the device from becoming "Unavailable" (#37607)

frenck committed:
Upgrade debugpy to 1.0.0b12 (#37599)

frenck committed:
Upgrade coverage to 5.2 (#37598)

frenck committed:
Fix typos in Hue integration (#37597)

frenck committed:
Add current temperature as separate sensor in Toon (#37336)

* Add current temperature as seperate sensor in Toon

* Set default enabled to False

* Trigger build

TheLastGimbus committed:
Add host names in esphome logs (#37587)

vangorra committed:
Remove withings use of deprecated classes (#37611)

Adjusting in bed device class to occupancy.

frenck committed:
Fix sync/async and small improvements to forked_daapd (#37619)

frenck committed:
Remove dead code from cast (#37620)

frenck committed:
Fix acmeda syn/async cover methods (#37618)

frenck committed:
Fix sync/async override in sms (#37621)

elupus committed:
Switch what is used for unique identifier (#37581)

* Switch what is used for device identifier

type_string can be changed if new device types are added based
on same driver.

* No need to slugify unique id

brefra committed:
Change MediaPlayerDevice into MediaPlayerEntity (#37629)

marcelveldt committed:
Ozw climate fixes (#37560)

* fix presets and mode conversion

* fix mapping issues in ozw climate

* build mapping table in advance

* Copying a dict to a list copies the keys by default

Jc2k committed:
Bump aiohomekit to 0.2.41 (#37602)

pvizeli committed:
Update azure-pipelines-wheels.yml for Azure Pipelines

bdraco committed:
Fix homekit test mocking missed in loop changeover (#37628)

balloob committed:
Reduce log level of unknown discovered services (#37617)

homeassistant committed:
[ci skip] Translation update

prystupa committed:
fix erroneous dependency used by Bond integration (simplejson to json) (#37642)

balloob committed:
Mark the example values as strings because that's what we expect (#37640)

marcelveldt committed:
Fix ozw entities cleanup on node removal (#37630)

* fix hass cleanup on node removal

also detect removefailednode command to issue a full cleanup

* satisfy pylint

pvizeli committed:
Update azure-pipelines-wheels.yml for Azure Pipelines

raman325 committed:
bump pyvizio version (#37644)

bdraco committed:
Increase slow setup logging to warning level (#37635)

When I fixed verbose logging in #36444, I did not
realize this would mean almost nobody would
see this now.

pvizeli committed:
Update azure-pipelines-wheels.yml for Azure Pipelines

firstof9 committed:
Add OZW support for set_config_parameter service (#37523)

* Add support for set_config_parameter service

* Adjusted elif to if

* More if/else cleanup

* More if/else cleanup

* Less nesting

* End loop properly

* Added byte type

* Convert break to return

pvizeli committed:
Update azure-pipelines-wheels.yml for Azure Pipelines

emontnemery committed:
Modify cast tests to setup via cast integration (#37256)

* Improve tests

* Improve tests

* Don't start zeroconf from tests

alexhardwicke committed:
Give fan and remote components unique LED strings (#37605)

raman325 committed:
Vizio: when checking new host against existing config entry hosts, make check hostname aware (#37397)

* make ip check hostname aware

* add executor job for sync function doing IO and remove errant comma

* revert change to update new_data explicitly for options keys since it is already done later

* empty commit to retrigger CI

pilehave committed:
Add preset modes to Touchline (#36054)

* Added preset modes.

* Flake8 passed.

* New and cleaner version.

* isort fixed?

* OrderedDict removed, constant OPERATION_LIST removed.

* ClimateDevice changed to ClimateEntity

* Two methods replaced with constants.

* Update homeassistant/components/touchline/climate.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Update homeassistant/components/touchline/climate.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Update homeassistant/components/touchline/climate.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* All changes needed.

* Cleaned up

* Clean up

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

mdegat01 committed:
Updated influxdb-client dependency to 1.8.0 (#37396)

mdegat01 committed:
Check buckets/dbs for validity during Influx sensor startup (#37391)

* Check buckets/dbs for validity during sensor startup

* Empty array instead of none

bsmappee committed:
Update Smappee integration with proper solar, voltage and reactive entities (#37407)

bachya committed:
Fix missing Guardian service strings (#37659)

stshontikidis committed:
Add kwargs to send_magic_packet() service individually (#37387)

ctalkington committed:
Apply more suggestions from bond code review (#37592)

* apply more suggestions from bond code review

* Update cover.py

* Update test_cover.py

* Update test_cover.py

* Update test_cover.py

* Update cover.py

* Update utils.py

* Update test_cover.py

* Update test_utils.py

* Delete test_utils.py

* Update cover.py

* Update test_cover.py

* Update test_cover.py

emontnemery committed:
Set MQTT sensor to state unavailable when value expires (#36609)

scop committed:
Convert syncthru to config flow and native SSDP discovery (#36690)

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

engrbm87 committed:
Use "next_state" attr instead of "post_pending" for ArmDisarm trait (#37325)

firstof9 committed:
Add ozw usercode support (#37390)

homeassistant committed:
[ci skip] Translation update

firstof9 committed:
OZW Usercodes update services.yaml with examples (#37667)

Shulyaka committed:
Add humidifier support to prometheus (#37112)

* Add humidifier support to promethease

* add humidifier dependency

* handle dependencies

* wrong place

* applied suggestions from code review

* fix typo

* Revert "wrong place"

This reverts commit 58e509d767d484e43aa108ca896146e514b79c10.

* Revert "handle dependencies"

This reverts commit 74bd523e08a32cd906809929f139d6469610f808.

* Revert "add humidifier dependency"

This reverts commit 8840b378e53d28a86eed5f62a8126769d3f58afe.

jduquennoy committed:
Refactor Enocean part 1 (#35927)

* First step of an EnOcean integration refactoring, including code reorganisation and support of a setup config flow

* Moved title to root of strings file

* Fixed pre-commit checks failures

* Fixed linter errors

* Updated formatted string format in logs

* Removed leftover comment

* Multiple changes after PR change requests.
Using an import flow for yaml config, removed unnecessary logs, added proper unload in __init__ and EnOceanDongle
Replaced config state machine by several flows.
Serial port validity check done in the EnOceanDongle class asynchronously, removed unique ID from config flow
Multiple cosmetic changes

* Multiple changes after PR change requests

* Added variable to store default value, as setdefault was caught returning None when the empty dict literal was passed as an argument

* Literal used directly

* Added tests for EnOcean config flows, changed static methods to bundle methods for bundle

* Updated variable name

* Added missing mock to test, replaced repeated magic strings by constants

* Changed imports to avoid an unused import warning from pylint on DOMAIN

* Adding pylint exception for unused import

* Added proper propagation of setup and unload to platforms, removed dead code, some syntax changes

* Removed setup_entry forwarding as the entities can only be configured using yaml

* Removed forwarding of unload

* Enabled code coverage for config flow only

* Clean up coveragerc

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

cgtobi committed:
Add back Netatmo public weather sensors (#34401)

* Add public weather sensors back in

* Remove stale code

* Cleanup after before adding entities

* Fix pylint complaint

* Add test for options flow

* Change mode to listbox

* Update .coveragerc

* Address comments

* Don't process empty list

* Address comment

* Fix mistake

* Make signal unique

* Make string more unique

* Fix merge conflict

elupus committed:
Split handling and application of event (#37665)

This way _handle_event can contain things available
when entity has been added to home assistant,
and _apply event can remain internal and used on init.

pvizeli committed:
Python 3.8 on core Container (#37677)

bsmappee committed:
Smappee dependency update (#37680)

elupus committed:
Detect lingering threads after tests (#37270)

* Detect lingering threads after tests

* Make sure cast is setup before checking state

* Make sure we ask executors of old hass to shutdown

We are not waiting here, just hoping for the best

* Make sure all instances of hass and executors is stopped.

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

* Also apply hass stopping to scripts

* Adjust to changes how we set up executor

* Add new CoreState.stopped

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

HarryJHolmes committed:
Change audio sample rate for apple watch homekit camera (#37637)

* Update type_cameras.py

* Apply suggestions from code review

Support both 24kHz and 16kHz sample rate

Co-authored-by: J. Nick Koston <nick@koston.org>

* Adjust formatting

* Reformat

Co-authored-by: J. Nick Koston <nick@koston.org>

gmta committed:
Round time values in get_age() to better approximate the actual age (#37125)

prystupa committed:
Add bond cover assumed state and local polling (#37666)

* Declare Bond covers as having assumed state, setup local polling for state updates

* Declare Bond covers as having assumed state, setup local polling for state updates (apply feedback from PR review)

* Declare Bond covers as having assumed state, setup local polling for state updates (apply feedback from PR review)

* Declare Bond covers as having assumed state, setup local polling for state updates (apply feedback from PR review)

homeassistant committed:
[ci skip] Translation update

bachya committed:
Actually fix Guardian entity services (#37700)

* Actually fix Guardian entity services

* Bump aioguardian

* Remove upgrade_firmware defaults

mdegat01 committed:
Revert "Updated influxdb-client dependency to 1.8.0" (#37396)" (#37697)

This reverts commit 9964bd40ed20c716ea0e960204061862c32fd479.

balloob committed:
Upgrade foobot-async (#37706)

elupus committed:
Rewrite rfxtrx init logic to do away with global object (#37699)

* Rewrite init logic to do away with global object

* Put constant at end

* Use a set instead of list for device_ids

PaulAnnekov committed:
bump tuyaha 0.0.7 (#37709)

djpremier committed:
Fix get profiles checking if has ptz capabilities (#37176)

bednar committed:
Update influxdb-client dependency to 1.8.0, fix test write for InfluxDB v2 (#37710)

* chore: updated influxdb-client dependency to 1.8.0, fixed testing write for InfluxDB v2

* fix: code style

pavoni committed:
Fix loopenergy callback updating HA before the object is initialised (#37650)

* Fix loopenergy callback updating HA before the object is initialised.

* Change to use async_added_to_hass.

balloob committed:
Fix Hue homekit discovery (#37694)

Co-authored-by: Franck Nijhof <git@frenck.dev>

emontnemery committed:
Tweak MQTT availability (#37719)

emontnemery committed:
Fix MQTT availability startup race (#37718)

pnbruckner committed:
Add new repeat loop for scripts and automations (#37589)

elupus committed:
Add rfxtrx device classes to known types (#37698)

* Add device classes to known types

* Update homeassistant/components/rfxtrx/sensor.py

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

bachya committed:
Re-add ability to use remote files (by URL) in Slack messages (#37161)

* Re-add remote file support for Slack

* More work

* Ensure Slack can only upload files from whitelisted directories

* Cleanup

* Finish work

* Code review

* Messing around

* Final cleanup

* Add comment explaining why we use aiohttp for remote files

* Typo

bdraco committed:
Use the shared zeroconf instance for homekit_controller (#37691)

* Use the shared zeroconf instance for homekit_controller

* bump version

* Update for upstream changes. Thank you @Jc2k !

* naming

* bump version

* empty commit to reset CI as re-run is not working

rohankapoorcom committed:
Add support for the DataUpdateCoordinator to not automatically update (#37734)

balloob committed:
Uninstall typing (#37735)

pnbruckner committed:
Remove legacy script mode and simplify remaining modes (#37729)

homeassistant committed:
[ci skip] Translation update

prystupa committed:
Support Fan domain in Bond integration (#37703)

* Support Fan domain in Bond integration

* Support Fan domain in Bond integration

* Support Fan domain in Bond integration (apply PR feedback)

* Support Fan domain in Bond integration (apply PR feedback)

prystupa committed:
Fix incorrect comparison of speed "off" by identity instead of by value (#37738)

prystupa committed:
Refactor Bond integration to remove duplication (#37740)

* Refactor Bond integration to remove duplication in Entity code and unit tests

* Refactor Bond integration to remove duplication in Entity code and unit tests (PR feedback)

haemishkyd committed:
Updates to poolsense integration (#37613)

* Created a binary sensor and corrected some review comments.

* Updated the poolsense class and its interface to handle credentials better

* Moved the client session to the PoolSense class.

* Apply suggestions from code review

* Update binary_sensor.py

* Update homeassistant/components/poolsense/__init__.py

* Update sensor.py

* Update binary_sensor.py

Co-authored-by: Chris Talkington <chris@talkingtontech.com>

balloob committed:
Bump ADS to 3.1.3 (#37748)

balloob committed:
Reference constraint files from requirement files (#37751)

* Reference constraint files from requirement files

* Update azure-pipelines-wheels.yml

Co-authored-by: Pascal Vizeli <pvizeli@syshack.ch>

fronzbot committed:
Bump blinkpy version to fix connection errors (#37755)

* Bump blinkpy version to fix connection errors

* Bump blinkpy version to fix connection errors

* Rebased, re-ran gen_requirements_all

balloob committed:
Bump pyHS100 to 3.5.1 (#37749)

pnbruckner committed:
Fix script queued mode typo (#37759)

homeassistant committed:
[ci skip] Translation update

prystupa committed:
Upgrade bond-home to 0.0.9 (#37764)

alandtse committed:
Bump teslajsonpy to 0.9.3. (#37771)

Fixes #37684

bdraco committed:
Significantly improve logging performance when no integrations are requesting debug level (#37776)

Kane610 committed:
UniFi - Handle session expiration (#37782)

prystupa committed:
Add Bond hub as a device for bond entities (#37772)

* Introduce Bond Hub concept

* Read Hub version information when setting up entry

* Link entities to Hub using via_device

* Add test to verify created Hub device properties

esev committed:
Add generic unavailable and last_updated metrics for prometheus (#37456)

* Add generic unavailable and last_updated metrics for prometheus

* Updated with feedback from the code review

elupus committed:
Switch rfxtrx to integration level config (#37742)

* Switch to integration level config

* Switch to per device config rather than per entity type

* All roller shutters should be added as covers

(there are non lighting types)

* Fixup tests that used invalid packets for platforms

* Avoid variable re-use

* Allow control events on sensors too

That way we get signal level sensors for these too

* Lint correction

* Don't filter sensors from config

Disable sensors from GUI if the entities are not wanted

* Correct usage of ATTR_ instead of CONF_

* Make sure the logging when a new entity is added includes the event

prystupa committed:
Add support for fan direction in bond integration (#37789)

* Add support for fan direction in bond integration

* Add support for fan direction (PR feedback)

haemishkyd committed:
Apply code quality updates to poolsense (#37781)

* Created a binary sensor and corrected some review comments.

* Updated the poolsense class and its interface to handle credentials better

* Moved the client session to the PoolSense class.

* Apply suggestions from code review

* Update binary_sensor.py

* Update homeassistant/components/poolsense/__init__.py

* Update sensor.py

* Update binary_sensor.py

* Removed unnecessary class variable

* Correcting a merge error.

Co-authored-by: Chris Talkington <chris@talkingtontech.com>

jjlawren committed:
Wrap possible I/O in executor (#37688)

Kane610 committed:
deCONZ - don't let light "attr" events update group data (#37797)

ajschmidt8 committed:
Fix Dockerfile.dev for VS Code devcontainer (#37801)

prystupa committed:
Add basic support for lights in bond integration (#37802)

homeassistant committed:
[ci skip] Translation update

elupus committed:
Replace rfxtrx entity events with integration events  (#37565)

* Drop per entity events

* Correct linting error

* Drop keys()

balloob committed:
Bump aiokafka to 0.6.0 (#37778)

elupus committed:
Drop dummy connection (#37805)

We now mock the module

coolsylver committed:
pydaikin version bump to 2.3.1: (#37682)

* Fix skyfi incorrect setting zone
* Add support for advanced modes for brp069 model

dermotduffy committed:
Allow an extra packet without dts (for Arlo camera streaming) (#37792)

* Allow 1 packet without dts. See https://github.com/twrecked/hass-aarlo/issues/151 .

* Reset boolean once a packet with dts is found.

* Fix no-else-continue lint error.

balloob committed:
Constraints pt3 (#37803)

jschlyter committed:
Add urlencode template filter (#37753)

* add urlencode template filter

* fix flake8

* add test to string ang integer

* better test vectors

elupus committed:
Add rfxtrx ability to send a raw command to device (#37793)

* Add the ability to send a raw command to device

* Add a test for new service

* Use async test

* Sort includes

balloob committed:
Drop white blacklist pt1 (#37816)

balloob committed:
Simplify logger integration (#37780)

Co-authored-by: J. Nick Koston <nick@koston.org>

Commit:
Add devolo binary sensor device class mapping (#37350)

* add device class mapping for binary sensors

* change if else statement to or

engrbm87 committed:
Properly set update_interval during Speedtest setup (#37708)

* Properly set update_interval during setup

* implement new update_interval method

tizzen33 committed:
Convert Toon expires_in value to float (#37716)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

prystupa committed:
Apply bond python related feedback from a prior PR (#37821)

elupus committed:
Switch rfxtrx to config entries (#37794)

* Switch to config flow setup

* Add minimal test for config flow

* Add myself as codeowner and address some review concerns

* Address some further review comments

gjbadros commented:
@rytilahti is your rewrite in yet?  If so, then this should get closed.  If not, we should still consider this to tide us over since the current code is clearly way too conservative in blocking every device when one is flakey.

rytilahti commented:
@gjbadros the rewrite is not yet in, and it may take a while. I just wrote a proposal in the WIP PR to separate the conversion to new backend to its own custom integration until it's ready to be released, that'd allow unblocking the PRs as long as someone steps up to maintain it.

MartinHjelmare commented:
@rytilahti Can we merge this?

@gjbadros please rebase on latest dev branch.

gjbadros committed:
Merge branch 'tplink_robustness_fix' of https://github.com/gjbadros/home-assistant into tplink_robustness_fix

gjbadros committed:
Merge branch 'tplink_robustness_fix' of https://github.com/gjbadros/home-assistant into tplink_robustness_fix

gjbadros commented:
Rebased. Let me know if there are any problems.  It'll be great to finally get this in.

MartinHjelmare commented:
The branch contains unrelated commits now. Those need to be removed.

gjbadros commented:
Just use the same branch name with _v2 appended.   I remade that one clean.

On Fri, Sep 18, 2020, 11:59 AM Martin Hjelmare <notifications@github.com>
wrote:

> The branch contains unrelated commits now. Those need to be removed.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/home-assistant/core/pull/35628#issuecomment-695034161>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AALOHTOGSTIGMYB4USHUQMDSGOUZLANCNFSM4NA4DRXA>
> .
>

MartinHjelmare commented:
Will you open a new PR?

gjbadros commented:
https://github.com/home-assistant/core/pull/40281

On Fri, Sep 18, 2020 at 5:29 PM Martin Hjelmare <notifications@github.com>
wrote:

> Will you open a new PR?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/home-assistant/core/pull/35628#issuecomment-695137435>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AALOHTMUQILQVY3WVMKQOMLSGP3PVANCNFSM4NA4DRXA>
> .
>
